wordpress-mobile/WordPress-Android
# Login rework: magic link request input screen

Introduces the Magic Link request screen.

Includes this functionality:
* Shows the Gravatar corresponding to the user email
* Performs the magiclink request
* Navigates to the "magiclink sent" screen when request succeeded
* Cancellable progress dialog. Note: the actual network is not cancelled and the magiclink email will be delivered to the user anyway. The dialog is basically dismissed and the screen won't change to the "sent" screen when dismissed.

The final visual touches, the messages and the final texts will be done in a future PR.

To test:
* On the Prologue screen, choose "LOG IN"
* Input an email address that us known to be registered on WordPress.com and hit "NEXT"
* Notice the screen change and shortly after the user Gravatar be shown
* Tap on "SEND LINK" and wait a few moments for the progress dialog to finish
* Notice the screen change to "magiclink sent" screen. Nothing is implemented in this screen though, except for the "Back" functionality.
* Check your email inbox for the magiclink email to have arrived
* That's all :)
